internal/controller/api_to_config.go

```diff
@@ -98,6 +98,11 @@ func apiToFRR(resources ClusterResources, alwaysBlock []net.IPNet) (*frr.Config,
 				return nil, err
 			}
 
+			// merging with itself to validate neighbor list
+			if _, err := mergeNeighbors(routerCfg.Neighbors, routerCfg.Neighbors); err != nil {
+				return nil, err
+			}
```

Comment:
let's wrap this in a `validateRouter` func, and also can we use `mergeRouterConfigs` instead?

Reply:
done